utxo/src/cache.rs

```diff
@@ -13,202 +13,35 @@
+};
 
 #[derive(Clone)]
 pub struct ConsumedUtxoCache {
-    container: BTreeMap<OutPoint, UtxoEntry>,
-    best_block: Id<GenBlock>,
-}
-
-pub trait FlushableUtxoView {
-    /// Performs bulk modification
-    fn batch_write(&mut self, utxos: ConsumedUtxoCache) -> Result<(), Error>;
-}
-
-// flush the cache into the provided base. This will consume the cache and throw it away.
-// It uses the batch_write function since it's available in different kinds of views.
-pub fn flush_to_base<T: FlushableUtxoView>(cache: UtxosCache, base: &mut T) -> Result<(), Error> {
-    base.batch_write(cache.consume())
+    pub(crate) container: BTreeMap<OutPoint, UtxoEntry>,
+    pub(crate) best_block: Id<GenBlock>,
 }
 
 #[derive(Clone)]
 pub struct UtxosCache<'a> {
     parent: Option<&'a dyn UtxosView>,
     current_block_hash: Id<GenBlock>,
-    utxos: BTreeMap<OutPoint, UtxoEntry>,
-    //TODO: do we need this?
+    pub(crate) utxos: BTreeMap<OutPoint, UtxoEntry>,
+    #[allow(dead_code)]
     memory_usage: usize,
 }
 
-/// Tells the state of the utxo
-#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
-#[allow(clippy::large_enum_variant)]
-pub enum UtxoStatus {
-    Spent,
-    Entry(Utxo),
-}
-
-impl UtxoStatus {
-    fn into_option(self) -> Option<Utxo> {
-        match self {
-            UtxoStatus::Spent => None,
-            UtxoStatus::Entry(utxo) => Some(utxo),
-        }
-    }
-}
-
-/// Just the Utxo with additional information.
-#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
-pub(crate) struct UtxoEntry {
-    status: UtxoStatus,
-    /// The utxo entry is dirty when this version is different from the parent.
-    is_dirty: bool,
-    /// The utxo entry is fresh when the parent does not have this utxo or
-    /// if it exists in parent but not in current cache.
-    is_fresh: bool,
-}
-
-impl UtxoEntry {
-    pub fn new(utxo: Utxo, is_fresh: bool, is_dirty: bool) -> UtxoEntry {
-        UtxoEntry {
-            status: UtxoStatus::Entry(utxo),
-            is_dirty,
-            is_fresh,
-        }
-    }
-
-    pub fn is_dirty(&self) -> bool {
-        self.is_dirty
-    }
-
-    pub fn is_fresh(&self) -> bool {
-        self.is_fresh
-    }
-
-    pub fn is_spent(&self) -> bool {
-        self.status == UtxoStatus::Spent
-    }
-
-    pub fn utxo(&self) -> Option<Utxo> {
-        match &self.status {
-            UtxoStatus::Spent => None,
-            UtxoStatus::Entry(utxo) => Some(utxo.clone()),
-        }
-    }
-
-    pub fn take_utxo(self) -> Option<Utxo> {
-        match self.status {
-            UtxoStatus::Spent => None,
-            UtxoStatus::Entry(utxo) => Some(utxo),
-        }
-    }
-
-    fn utxo_mut(&mut self) -> Option<&mut Utxo> {
-        match &mut self.status {
-            UtxoStatus::Spent => None,
-            UtxoStatus::Entry(utxo) => Some(utxo),
-        }
-    }
-}
-
 impl<'a> UtxosCache<'a> {
     #[cfg(test)]
     pub(crate) fn new_for_test(best_block: Id<GenBlock>) -> Self {
```

Comment:
Nitpick: currently the `#[cfg(test)]` attribute makes no sense together with `pub(crate)`.
(https://github.com/rust-lang/cargo/issues/8379)